src/compiler.jl

```diff
@@ -6101,7 +6103,24 @@ function to_tape_type(Type::LLVM.API.LLVMTypeRef)::Tuple{DataType,Bool}
         elseif N == 128
             return UInt128, false
         else
-            error("Can't construct tape type for integer of width $N")
+            T = get(WideTypes, N, nothing)
+            if isnothing(T)
+                symname = Symbol(:UInt,N)
+                emptysvec = Core.svec()
+                _module = Main
+                Tp = GC.@preserve emptysvec symname @ccall jl_new_primitivetype(symname::Symbol, 
```

Comment:
is there a reason we have to do this, rather than some sort of @ eval thing cc @vchuravy

Reply:
Because eval cannot be used in a generated function and we're inside one at this point. I guess the ccall shouldn't really be valid for the same reason. A bit stuck here unless we can go back to returning NTuple{N >> 6, UInt64} for N > 128. The only way to get to that branch in Julia is from new primitive types and there are no architectures that Julia supports where anything larger than i128 is supported so I don't think this impacts anything that has the possibility of performing arithmetic.